filter.imp

```diff
@@ -1,13 +1,119 @@
 [
+   # If an architecture does not provide its own copy of asm/foo.h, then it will get an
+   # asm-generic/foo.h which can still be included as asm/foo,h
+   # This list was generated from include/uapi/asm-generic/Kbuild
+   { include: ["@.asm-generic/auxvec.h.", private, "<asm/auxvec.h>", "public"] },
+   { include: ["@.asm-generic/bitsperlong.h.", private, "<asm/bitsperlong.h>", "private"] },
+   { include: ["@.asm-generic/bpf_perf_event.h.", private, "<asm/bpf_perf_event.h>", "public"] },
+   { include: ["@.asm-generic/byteorder.h.", private, "<asm/byteorder.h>", "public"] },
+   { include: ["@.asm-generic/errno.h.", private, "<asm/errno.h>", "private"] },
+   { include: ["@.asm-generic/fcntl.h.", private, "<asm/fcntl.h>", "public"] },
+   { include: ["@.asm-generic/ioctl.h.", private, "<asm/ioctl.h>", "public"] },
+   { include: ["@.asm-generic/ioctls.h.", private, "<asm/ioctls.h>", "public"] },
+   { include: ["@.asm-generic/ipcbuf.h.", private, "<asm/ipcbuf.h>", "public"] },
+   { include: ["@.asm-generic/mman.h.", private, "<asm/mman.h>", "public"] },
+   { include: ["@.asm-generic/msgbuf.h.", private, "<asm/msgbuf.h>", "public"] },
+   { include: ["@.asm-generic/param.h.", private, "<asm/param.h>", "public"] },
+   { include: ["@.asm-generic/poll.h.", private, "<asm/poll.h>", "public"] },
+   { include: ["@.asm-generic/posix_types.h.", private, "<asm/posix_types.h>", "public"] },
+   { include: ["@.asm-generic/ptrace.h.", private, "<asm/ptrace.h>", "public"] },
+   { include: ["@.asm-generic/resource.h.", private, "<asm/resource.h>", "private"] },
+   { include: ["@.asm-generic/sembuf.h.", private, "<asm/sembuf.h>", "public"] },
+   { include: ["@.asm-generic/setup.h.", private, "<asm/setup.h>", "public"] },
+   { include: ["@.asm-generic/shmbuf.h.", private, "<asm/shmbuf.h>", "public"] },
+   { include: ["@.asm-generic/sigcontext.h.", private, "<asm/sigcontext.h>", "public"] },
+   { include: ["@.asm-generic/siginfo.h.", private, "<asm/siginfo.h>", "public"] },
+   { include: ["@.asm-generic/signal(-defs)?.h.", private, "<asm/signal.h>", "private"] },
```

Comment:
signal-defs is out of place. it is not listed in include/uapi/asm-generic/Kbuild. break this line up to have -defs be another rule outside of this group.  Oh, it's listed below, so you just need to update this regex to remove the group.